src/screens/sync/index.tsx

```diff
@@ -28,21 +33,38 @@ const styles = StyleSheet.create({
 
 export const SyncScreen: ScreenComponent<"Sync"> = ({ route }) => {
   const [viewMode, setViewMode] = React.useState<ViewMode>("list");
+  const [modalContent, setModalContent] = React.useState<TitleAndDescription>({
+    title: "",
+    description: "",
+  });
+  const { sheetRef, openSheet } = useBottomSheetModal({ openOnMount: false });
+
+  function setAndOpenModal(titleAndDescription: TitleAndDescription) {
+    setModalContent(titleAndDescription);
+    openSheet();
+  }
 
   return (
-    <ScrollView
-      style={styles.container}
-      contentContainerStyle={styles.scrollContentContainer}
-    >
-      <ProjectInfo
-        name="Project Catapult"
-        viewMode={viewMode}
-        toggleViewMode={() => {
-          setViewMode((prev) => (prev === "list" ? "bubbles" : "list"));
-        }}
-      />
-      <Devices mode={viewMode} />
-    </ScrollView>
+    <React.Fragment>
+      <ScrollView
+        style={styles.container}
+        contentContainerStyle={styles.scrollContentContainer}
+      >
+        <ProjectInfo
+          name="Project Catapult"
+          viewMode={viewMode}
+          toggleViewMode={() => {
+            setViewMode((prev) => (prev === "list" ? "bubbles" : "list"));
+          }}
+        />
+        <Devices>
+          {viewMode === "list" ? (
+            <DeviceList setAndOpenModal={setAndOpenModal} />
+          ) : null}
```

Comment:
exposing the `DeviceList` here so that it can have access to setting the content of the modal, given that the different types of device lists change the content of the modal

Reply:
think surfacing `DeviceList` here is unnecessary and passing the relevant prop(s) through to `Device` then `DeviceList` is cleaner and keeps `DeviceList` as something only the `Device` needs to worry about.

suggestion would be to just pass the appropriate prop to `Devices` and let `Devices` handle its usage e.g.

```tsx
<Devices viewMode={viewMode} onInfoPress={setAndOpenModal} />
```